presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/converter/GlueToPrestoConverter.java

```diff
@@ -69,7 +70,9 @@ public static Table convertTable(com.amazonaws.services.glue.model.Table glueTab
                 .setDatabaseName(dbName)
                 .setTableName(glueTable.getName())
                 .setOwner(nullToEmpty(glueTable.getOwner()))
-                .setTableType(glueTable.getTableType())
+                // Athena Tables work fine with empty table type but this would NPE on Presto.
+                // if no table type has been set, assume MANAGED_TABLE.
+                .setTableType(firstNonNull(glueTable.getTableType(), TableType.MANAGED_TABLE.name()))
```

Comment:
Does Athena assume MANAGED as well? I.e. does it remove the data when you drop such a table?

Reply:
Would it be possible to have a test in `TestHiveGlueMetastore`?

Reply:
Wil add a test.

Athena is not dropping the data, just the metadata for a table without table type.

When drop the managed table in presto, it does not delete the data files either (I have a S3 directory location in Glue, not a list of files).

Reply:
When using thrift metastore, HMS is responsible for deleting data when MANAGED table is being dropped.

When using Glue, the data for MANAGED table seems to get deleted here:
https://github.com/prestosql/presto/blob/b066f1843bed2b192eeed7c0ebcd71f08664c196/presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java#L490-L492

> Athena is not dropping the data, just the metadata for a table without table type.

So IMO it should default to EXTERNAL_TABLE.

Reply:
Agreed. The Athena documentation explicitly says that all tables are external: https://docs.aws.amazon.com/athena/latest/ug/drop-table.html

> Removes the metadata table definition for the table named `table_name`. When you drop an external table, the underlying data remains intact because all tables in Athena are `EXTERNAL`.